File: Medium/3147. Taking Maximum Energy From the Mystic Dungeon/solution.py
class Solution:
    def maximumEnergy(self, energy: List[int], k: int) -> int:
        n = len(energy)
        rev = energy[::-1]

        for i in range(k, n):
            rev[i] = rev[i] + rev[i - k]

        res = float("-inf")
        for i in range(n):
            res = max(res, rev[i])

        return res


# Can abbreviated as:

class Solution:
    def maximumEnergy(self, energy: List[int], k: int) -> int:
        n = len(energy)

        res = float("-inf")
        for i in range(n - 1, -1, -1):
            if i < n - k:
                energy[i] = energy[i] + energy[i + k]
            res = max(res, energy[i])

        return res

File: Medium/3147. Taking Maximum Energy From the Mystic Dungeon/solution.cs
public class Solution {
    public int MaximumEnergy(int[] energy, int k) {
        int n = energy.Length;
        int res = Int32.MinValue;

        for(int i = n - 1; i >= 0; i--){
            if(i < n - k){
                energy[i] = energy[i] + energy[i + k];
            }
            res = Math.Max(res, energy[i]);
        }

        return res;
    }
}